client/state/sites/plans/connections.js

```diff
@@ -41,25 +40,21 @@ export function connectCancelSitePlanTrial( action, dispatch ) {
 	} );
 }
 
-export function connectFetchSitePlans( action, dispatch ) {
-	return new Promise( ( resolve ) => {
-		wpcom.undocumented().getSitePlans( action.siteId, ( error, data ) => {
-			if ( error ) {
-				debug( 'Fetching site plans failed: ', error );
-
-				const errorMessage = error.message || i18n.translate( 'There was a problem fetching site plans. Please try again later or contact support.' );
-
-				dispatch( {
-					type: SITE_PLANS_FETCH_FAILED,
-					siteId: action.siteId,
-					error: errorMessage
-				} );
-			} else {
-				dispatch( fetchSitePlansCompleted( action.siteId, data ) );
-			}
-
-			resolve();
-		} );
-	} );
-}
+export const connectFetchSitePlans = wpcom => ( action, dispatch ) =>
+	wpcom.undocumented().getSitePlans( action.siteId, ( error, data ) => {
```

Comment:
What about having `wpcom.undocumented().getSitePlans` passed instead of `wpcom` param? That would make it super easy to test and much more flexible, as we weren't tied to `wpcom` implementation details that much here. I'm not sure myself if that makes any sense, but leaving a note to think about :D